File: week1/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        strs.sort(key=lambda x:len(x))
        ans=strs[0]
        for i in range(len(strs)):
            for j in range(len(strs[i])):
                if j>=len(ans):
                    break
                if strs[i][j]!=ans[j]:
                    ans=ans[:j]
        return ans 

File: week1/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ans="";
        if(strs.size()==1)
        return strs[0];
        int j=0;
        int c=0;
        for(int i=0; i>=0; i++){
            for( j=0; j<strs.size()-1; j++){
                if(strs[j]=="")
                return ans;
                if(i==strs[j].size()-1)
                     c++;
                if(strs[j][i]!=strs[j+1][i])
                  return ans;
            } 
            if(j==0)
            break;
            ans+=strs[j][i];
           
            if(c!=0)
            break;
        }
        return ans;
    }
};